openmc/deplete/microxs.py

```diff
@@ -20,168 +22,173 @@
-        if isinstance(domain, openmc.Material):
-            domain_filter = openmc.MaterialFilter([domain])
-        elif isinstance(domain, openmc.Cell):
-            domain_filter = openmc.CellFilter([domain])
-        elif isinstance(domain, openmc.Universe):
-            domain_filter = openmc.UniverseFilter([domain])
-        else:
-            raise ValueError(f"Unsupported domain type: {type(domain)}")
-
-        rr_tally = openmc.Tally(name='MicroXS RR')
-        rr_tally.filters = [domain_filter, energy_filter]
-        rr_tally.nuclides = nuclides
-        rr_tally.multiply_density = False
-        rr_tally.scores = reactions
-
-        flux_tally = openmc.Tally(name='MicroXS flux')
-        flux_tally.filters = [domain_filter, energy_filter]
-        flux_tally.scores = ['flux']
-        tallies = openmc.Tallies([rr_tally, flux_tally])
-
-        model.tallies = tallies
-
-        # create temporary run
-        with tempfile.TemporaryDirectory() as temp_dir:
-            if run_kwargs is None:
-                run_kwargs = {}
-            run_kwargs.setdefault('cwd', temp_dir)
-            statepoint_path = model.run(**run_kwargs)
-
-            with StatePoint(statepoint_path) as sp:
-                rr_tally = sp.tallies[rr_tally.id]
-                rr_tally._read_results()
-                flux_tally = sp.tallies[flux_tally.id]
-                flux_tally._read_results()
-
-        # Get reaction rates and flux values
-        reaction_rates = rr_tally.mean.sum(axis=0)  # (nuclides, reactions)
-        flux = flux_tally.mean[0, 0, 0]
-
-        # Divide RR by flux to get microscopic cross sections
-        xs = reaction_rates / flux
-
-        # Build Series objects
-        series = {}
-        for i, rx in enumerate(reactions):
-            series[rx] = Series(xs[..., i], index=rr_tally.nuclides)
-
-        # Revert to the original tallies and materials
-        model.tallies = original_tallies
-
-        return cls(series).rename_axis('nuclide')
-
-    @classmethod
-    def from_array(cls, nuclides, reactions, data):
-        """
-        Creates a ``MicroXS`` object from arrays.
+            raise DataError(
+                "No depletion chain specified and could not find depletion "
+                "chain in openmc.config['chain_file']"
+            )
+    chain = Chain.from_xml(chain_file)
+    if reactions is None:
+        reactions = chain.reactions
+    if not nuclides:
+        cross_sections = _find_cross_sections(model)
+        nuclides_with_data = _get_nuclides_with_data(cross_sections)
+        nuclides = [nuc.name for nuc in chain.nuclides
+                    if nuc.name in nuclides_with_data]
+
+    # Set up the reaction rate and flux tallies
+    if energies is None:
+        energies = [0.0, 100.0e6]
+    if isinstance(energies, str):
+        energy_filter = openmc.EnergyFilter.from_group_structure(energies)
+    else:
+        energy_filter = openmc.EnergyFilter(energies)
```

Comment:
```suggestion
    if energies is None:
        energy_filter = openmc.EnergyFilter([0.0, 100.0e6])
    elif isinstance(energies, str):
        energy_filter = openmc.EnergyFilter.from_group_structure(energies)
    else:
        energy_filter = openmc.EnergyFilter(energies)
```
I think making use of an ```elif``` here could remove one check in some case